### DISCUSSION TITLE
Catch All Domain in one Project

### DISCUSSION THREAD

**Discussion by luiswagener** [coollabsio/coolify#2997] [Q&A]
Hello,

I'm trying to use Coolify to run my multi-tenant SaaS application, but I'm having trouble setting up a wildcard domain within a single project. My project needs to handle subdomains without requiring a restart of the application.
Is there a way to add this to my project or is it not possible at the moment?

**Comment by Eskelia** [comment]
Hi,

You should check this guide : https://coolify.io/docs/knowledge-base/traefik/wildcard-certificates

I'm also trying to use Coolify for a multi-tenatnt SaaS app. I managed to configure dynamic subdomains using the guide, but I can't generate certificates for custom domains. (My users can also link a custom domain by creating an A record pointing to the VPS IP)

Is there a way to generate certificates for dynamic custom domains ?

**Reply by yewyewxd** [reply]
I have followed the guide too, but could not set it up for my Next.js app. Whenever I go to my `domain.com` or `subdomain.domain.com`, it shows a 404 page (SSL certified). Could you please provide the steps that you took to achieve those?

**Reply by da411d** [reply]
@yewyewXD same problem

**Reply by sbushell92** [reply]
Same problem

**Reply by djsisson** [reply]
for wildcard domains, the only way to generate certificates is either using dns challenge or installing the wildcard certificate yourself

you can't use the default method of a http challenge, since the sub domain needs to be pre known

https://doc.traefik.io/traefik/https/acme/#dnschallenge

**Reply by kilobyte2007** [reply]
What if I don't want to generate the certificates, as those would be handled by Cloudflare and I just want the app to accept any domain? Is it possible?

**Reply by djsisson** [reply]
When you say any domain, do you mean any of your own or actually any domain?

If its the latter you would need to use ssl for saas https://blog.cloudflare.com/introducing-ssl-for-saas/

But although cf terminates the initial request, you should not be then sending plain HTTP from cloudflare across the Internet to your server.

You can just use dns wildcard challenge or add the cf origin wildcard that they give you this would mean https to your server, this works fine using full mode. For strict you would need to use cf api to generate a cf origin per customer domain.

Since traffic is sent to the fallback domain you specify with the customer's host header. And your cf origin cert satisfies ssl with cf, you just add a blank host regexp in traefik to send all domains to your app

**Reply by SamBabarinde** [reply]
I haven't been able to solve this for months now, main domain works but subdomains shows plain 404 page
